src/python/pants/backend/python/subsystems/repos.py

```diff
@@ -27,6 +27,17 @@ class PythonRepos(Subsystem):
             corresponds to the `--find-links` option.
             """
         ),
+        removal_version="2.16.0.dev0",
```

Comment:
I did a longer release cycle for `[python-repos]` than `[python].manylinux` because we knew many users are setting these 2 options. cc @jriddy

Reply:
What is the maintenance burden of keeping it? I don't have the cognitive load free to wade too deeply into this particular issue, but based on what I'm reading I'd argue that there's not a particularly good reason to get rid of it at all:

> Using the old options is equivalent to using __default__ with the new options. We ban setting both options at the same time to avoid ambiguity. So, we respect old behavior.

Sounds like this can live together and become a "buried" option in perpetuity.